src/lib/zkapps_examples/big_circuit/zkapps_big_circuit.ml

```diff
@@ -1,29 +1,25 @@
 open Signature_lib
+open Snark_params.Tick.Run
+open Mina_base
 
 (** Constructs a circuit with [num_constraints] constraints. *)
-let rec big_circuit (num_constraints : int) =
-  let open Snark_params.Tick.Run in
-  let b = exists Field.typ ~compute:(fun () -> Field.Constant.one) in
-  Field.Assert.equal b (constant Field.typ Field.Constant.one) ;
-  if next_auxiliary () > num_constraints then ()
-  else big_circuit num_constraints
-
-(* Make sure that the helper works. *)
-let () =
-  let num_constraints =
-    Snark_params.Tick.Run.constraint_count (fun _ -> big_circuit 10)
-  in
-  assert (num_constraints = 10)
+let big_circuit num_constraints =
+  for _ = 0 to num_constraints do
+    let b = exists Field.typ ~compute:(fun () -> Field.Constant.one) in
+    Field.Assert.equal b (constant Field.typ Field.Constant.one)
+  done ;
+  Format.printf "constraints: %d, wanted: %d\n" (next_auxiliary ())
+    num_constraints
```

Comment:
I still don't understand why the recursive code on the left did not work out (except when using small values for the number of constraints), I believe the vector of input was not long enough during witness generation (but I'm not sure I understand why):

```
Adding a test to check that creating a circuit too large in pickles throws an exception.

The exception:

Uncaught exception:
  
  (monitor.ml.Error (Failure vector_get)
   ("Raised by primitive operation at Kimchi_pasta_basic.Fp in file \"src/lib/crypto/kimchi_backend/pasta/basic/kimchi_pasta_basic.ml\", line 79, characters 24-140"
    "Called from Kimchi_backend_common__Plonk_constraint_system.Make.compute_witness.(fun) in file \"src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml\", line 461, characters 43-62"
    "Called from Stdlib__Array.iteri in file \"array.ml\", line 129, characters 31-51"
    "Called from Base__List.iteri.(fun) in file \"src/list.ml\", line 544, characters 7-12"
    "Called from Stdlib__List.fold_left in file \"list.ml\", line 121, characters 24-34"
    "Called from Base__List0.fold in file \"src/list0.ml\" (inlined), line 21, characters 22-52"
    "Called from Base__List.iteri in file \"src/list.ml\", line 543, characters 5-62"
    "Called from Kimchi_backend_common__Plonk_constraint_system.Make.compute_witness in file \"src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml\", line 454, characters 4-597"
    "Called from Kimchi_pasta__Vesta_based_plonk.Proof.Backend.create_aux in file \"src/lib/crypto/kimchi_backend/pasta/vesta_based_plonk.ml\", line 102, characters 8-68"
    "Called from Kimchi_backend_common__Plonk_dlog_proof.Make.create_async in file \"src/lib/crypto/kimchi_backend/common/plonk_dlog_proof.ml\", line 393, characters 6-70"
    "Called from Pickles__Step.Make.f.(fun) in file \"src/lib/pickles/step.ml\", line 746, characters 22-266"
    "Called from Snarky_backendless__Snark0.Run.Make_basic.finalize_is_running in file \"src/lib/snarky/src/base/snark0.ml\", line 1627, characters 14-18"
    "Called from Pickles__Step.Make.f.(fun) in file \"src/lib/pickles/step.ml\", line 729, characters 6-1023"
    "Called from Pickles.Make_str.Make.compile.f.wrap in file \"src/lib/pickles/pickles.ml\", line 811, characters 14-59"
    "Called from Pickles.Make_str.compile.adjust_provers.(fun) in file \"src/lib/pickles/pickles.ml\", line 1178, characters 32-62"
    "Called from Zkapps_examples.compile.go.prover in file \"src/lib/zkapps_examples/zkapps_examples.ml\", line 578, characters 14-32"
    "Called from Async_kernel__Monitor.Exported_for_scheduler.schedule'.upon_work_fill_i in file \"src/monitor.ml\", line 295, characters 42-51"
    "Called from Async_kernel__Job_queue.run_job in file \"src/job_queue.ml\" (inlined), line 128, characters 2-5"
    "Called from Async_kernel__Job_queue.run_jobs in file \"src/job_queue.ml\", line 169, characters 6-47"
    "Caught by monitor block_on_async"))

Raised at Base__Result.ok_exn in file "src/result.ml" (inlined), line 201, characters 17-26
Called from Async_unix__Thread_safe.block_on_async_exn in file "src/thread_safe.ml", line 132, characters 29-63
Called from Async_unix__Thread_safe.block_on_async_exn in file "src/thread_safe.ml" (inlined), line 181, characters 27-54
Called from Dune__exe__Big_circuit.wrapped in file "src/lib/transaction_snark/test/zkapps_examples/big_circuit/big_circuit.ml", line 59, characters 6-49
```

the for loop seems to work without issues,